src/BaseModel.js

```diff
@@ -1,32 +1,39 @@
 /**
-* Base class for all models
-*/
+ * Base class for all models
+ */
 export default class BaseModel {
 	/**
 	 * Constructor for the base model.
-	 * @param {string} id see [BaseModel's id property]{@link BaseModel#id}
-	 * @param {Date} createdAt see [BaseModel's createdAt property]{@link BaseModel#createdAt}
-	 * @param {Date} updatedAt see [BaseModel's updatedAt property]{@link BaseModel#updatedAt}
+	 * @param {object} options
+	 * @param {string} options.id see [BaseModel's id property]{@link BaseModel#id}
+	 * @param {Date} options.createdAt see [BaseModel's createdAt property]{@link BaseModel#createdAt}
+	 * @param {Date} options.updatedAt see [BaseModel's updatedAt property]{@link BaseModel#updatedAt}
 	 */
-	constructor(id, createdAt, updatedAt) {
+	constructor(options = {}) {
+		if (options instanceof BaseModel) {
```

Comment:
BaseModel is an abstract class, so we don't actually need that, we can have the `{...}` right in the constructor. In fact, it may make sense to [enforce it being an abstract class by checking `new.target`](https://programmer.ink/think/use-es6-new-target-to-simulate-abstract-classes.html)